include/COLA.hh

```diff
@@ -195,36 +204,68 @@ namespace cola {
         void regConv(std::shared_ptr<VFactory>&& factory, const std::string& name){converterMap.emplace(name, std::move(factory));}
         void regWrite(std::shared_ptr<VFactory>&& factory, const std::string& name){writerMap.emplace(name, std::move(factory));}
-    };
 
-    class ColaRunManager {
-    public:
-        explicit ColaRunManager(FilterAnsamble&& ansamble) : filterAnsamble(std::move(ansamble)) {}
-        ~ColaRunManager() = default;
-        void run();
-        void test();
-    private:
-        FilterAnsamble filterAnsamble;
+        MetaData parseStrToMeta(std::string data);
     };
 
 
-    inline cola::FilterAnsamble cola::MetaProcessor::parse(const cola::MetaData& data) {
+    inline cola::FilterAnsamble cola::MetaProcessor::parse(const cola::SimpleMetaData& data) {
+        FilterAnsamble ansamble;
+        ansamble.generator = std::shared_ptr<VGenerator>(dynamic_cast<VGenerator*>(generatorMap.at(data.generatorName)->create("")));
+        ansamble.converters.push(std::shared_ptr<VConverter>(dynamic_cast<VConverter*>(converterMap.at(data.converterName)->create(""))));
+        ansamble.writer = std::shared_ptr<VWriter>(dynamic_cast<VWriter*>(writerMap.at(data.writerName)->create("")));
+        return ansamble;
+    }
+
+    inline cola::FilterAnsamble cola::MetaProcessor::parse(std::string data) {
         FilterAnsamble ansamble;
-        ansamble.generator = std::shared_ptr<VGenerator>(dynamic_cast<VGenerator*>(generatorMap.at(data.generatorName)->create()));
-        ansamble.converters.push(std::shared_ptr<VConverter>(dynamic_cast<VConverter*>(converterMap.at(data.converterName)->create())));
-        ansamble.writer = std::shared_ptr<VWriter>(dynamic_cast<VWriter*>(writerMap.at(data.writerName)->create()));
+        MetaData meta = parseStrToMeta(data);
+        ansamble.generator = std::shared_ptr<VGenerator>(dynamic_cast<VGenerator*>(generatorMap.at(meta.generatorName)->create(meta.filterParamMap.at(meta.generatorName))));
+        while(!meta.converterNames.empty())
+        {ansamble.converters.push(std::shared_ptr<VConverter>(dynamic_cast<VConverter*>(converterMap.at(meta.converterNames.front())->create(meta.filterParamMap.at(meta.converterNames.front())))));
+            meta.converterNames.pop();}
+        ansamble.writer = std::shared_ptr<VWriter>(dynamic_cast<VWriter*>(writerMap.at(meta.writerName)->create(meta.filterParamMap.at(meta.writerName))));
         return ansamble;
     }
 
+    inline cola::MetaData cola::MetaProcessor::parseStrToMeta(std::string data) {
+        std::vector<std::string> filters; cola::MetaData metaData;
+        boost::split(filters, data, boost::is_any_of("\n"));
+        for(int flt = 0; flt < filters.size(); ++flt){
+            std::vector<std::string> tmp;
+            boost::split(tmp, filters.at(flt), boost::is_any_of(" "));
+            std::string res = "";
+            for(int i = 1; i < tmp.size(); ++i){res+=tmp.at(1);}
```

Comment:
consider using iterators (std::vector<T>::iterator) or range-based loops. Thery are a part of c++11 standart and are bset practices when working with STL containers

Reply:
It also allows to avoid blunders like .at(1) instead of at(i) ;)

Reply:
I also actually don't get the purpose of this. Why should we remove all spaces from a substring for a filter? How should a Factory parse it? Isn't it actually up to VFactory implementation to decide what to do with a substring?

Reply:
It is a bug.

Reply:
fixed